sgv2-docsapi/src/main/java/io/stargate/sgv2/docsapi/service/schema/JsonSchemaManager.java

```diff
@@ -0,0 +1,190 @@
+                Iterator<ProcessingMessage> it = report.iterator();
+                while (it.hasNext()) {
+                  ProcessingMessage msg = it.next();
+                  msgs += String.format("[%s]: %s; ", msg.getLogLevel(), msg.getMessage());
+                }
+                Throwable failure =
+                    new ErrorCodeRuntimeException(ErrorCode.DOCS_API_JSON_SCHEMA_INVALID, msgs);
+                return Uni.createFrom().failure(failure);
+              }
+            });
+  }
+
+  /**
+   * Validates a JSON document against a given table's schema
+   *
+   * @param table the table that has a schema
+   * @param document the document, as stringified JSON
+   * @return a Uni with Boolean detailing whether or not the document complies with the schema.
+   */
+  public Uni<Boolean> validateJsonDocument(Uni<Schema.CqlTable> table, String document) {
+    return table
+        .onItem()
+        .ifNotNull()
+        .transform(
+            t -> {
+              String comment = t.getOptionsMap().getOrDefault("comment", null);
+              if (comment == null) {
+                // If there is no valid schema, then the document is valid
+                return true;
+              }
+              JsonNode jsonSchema;
+              try {
+                jsonSchema = objectMapper.readTree(comment);
+              } catch (JsonProcessingException e) {
+                // If there is no valid schema, then the document is valid
+                return true;
+              }
+
+              try {
+                validate(jsonSchema.get("schema"), document);
+              } catch (ProcessingException e) {
+                return true;
+              }
+              return true;
+            });
+  }
+
+  /**
+   * Validates a JSON document against a given table's schema
+   *
+   * @param table the table that has a schema
+   * @param document the document, as JsonNode
+   * @return a Uni with Boolean detailing whether or not the document complies with the schema.
+   */
+  public Uni<Boolean> validateJsonDocument(Uni<Schema.CqlTable> table, JsonNode document) {
+    return table
+        .onItem()
+        .ifNotNull()
+        .transform(
+            t -> {
+              String comment = t.getOptionsMap().getOrDefault("comment", null);
+              if (comment == null) {
+                // If there is no valid schema, then the document is valid
+                return true;
+              }
+              JsonNode jsonSchema;
+              try {
+                jsonSchema = objectMapper.readTree(comment);
+              } catch (JsonProcessingException e) {
+                // If there is no valid schema, then the document is valid
+                return true;
+              }
+
+              try {
+                validate(jsonSchema.get("schema"), document);
+              } catch (ProcessingException e) {
+                return true;
+              }
+              return true;
+            });
+  }
```

Comment:
wait wait wait, why not going through the `getJsonSchema`, we don't want to double the same code:

```java
return getJsonSchema(table)
    .map(schema -> {
        if (null == schema) {
           return true:
        }     
        
       try {
          validate(jsonSchema.get("schema"), document);
        } catch (ProcessingException e) {
          return true;
        }
        return true;
    }
```